rebasebot/bot.py

```diff
@@ -506,13 +513,26 @@ def run(
         with open(cloner_credentials, "w", encoding='utf-8') as cloner_credentials_file:
             cloner_credentials_file.write(gh_cloner_app.session.auth.token)
 
+    # list with all the pr's having the label rebase/manual
+    prs_with_manual_rebase_label = []
+
     try:
         dest_repo = gh_app.repository(dest.ns, dest.name)
         logging.info("Destination repository is %s", dest_repo.clone_url)
         rebase_repo = gh_cloner_app.repository(rebase.ns, rebase.name)
         logging.info("rebase repository is %s", rebase_repo.clone_url)
         source_repo = gh_app.repository(source.ns, source.name)
         logging.info("source repository is %s", source_repo.clone_url)
+
+        prs = rebase_repo.pull_requests()
+        for pull_req in prs:
+            for label in pull_req.labels:
+                if label['name'] == 'rebase/manual':
+                    prs_with_manual_rebase_label.append(pull_req)
```

Comment:
Should we just send a slack message and return early?

Reply:
I'm still thinking that it would be better to return early here.
I think it would also be good to isolate this check for manual rebase prs.

Reply:
I think that I can get rid of most of the code here. It is enough to just `return False` once we encounter a PR with such label. What can happen in that case is:

1.  We don't find any PR with label `rebase/manual` and we continue as usual.
2. We do find a PR, doesn't matter whether there is just one or multiple - we just return earlier as mentioned before and send the slack message.

Therefore we can get rid of the `prs_with_manual_rebase_label` list and also the checks for such PRs in functions `_is_push_required` and `_is_pr_available`. What do you think about this thought, @lobziik?

Reply:
`return False` will cause non-zero exit code which then will cause our periodics to fail, i would rather go with `return True` since it's an expected situation.

But yeah, agree with the other.

Another small thing, perhaps, it would also be worth wrapping this thing into a separate function, something like
```
def _manual_rebase_in_dest(dest_repo) -> bool:
      ...
```

with the actual prs check inside.

Reply:
Updated the code with the aforementioned changes.